l10n_ch_import_cresus/wizard/l10n_ch_import_cresus.py

```diff
@@ -0,0 +1,344 @@
+                and all
+                 Balance sheet account must have a deferral \
+                method = 'balance'. </br>'''
+                                )
+
+                 }
+
+    HEAD_CRESUS = ['date', 'debit', 'credit', 'pce',
+                   'ref', 'amount', 'typtvat', 'currency_amount',
+                   'analytic_account']
+    HEAD_OPENERP = ['ref', 'date', 'period_id', 'journal_id',
+                    'line_id/account_id', 'line_id/partner_id', 'line_id/name',
+                    'line_id/debit', 'line_id/credit',
+                    'line_id/account_tax_id',
+                    'line_id/analytic_account_id']
+    MOVE_IDS = []
+
+    def open_account_moves(self):
+        move_ids = self.MOVE_IDS
+        res = {
+            'domain': str([('id', 'in', move_ids)]),
+            'name': 'Account Move',
+            'view_type': 'form',
+            'view_mode': 'tree,form',
+            'res_model': 'account.move',
+            'view_id': False,
+            'type': 'ir.actions.act_window',
+        }
+        return res
+
+    def format_messages(self, messages):
+        """Format error messages generated by the BaseModel.load method
+
+        :param messages: return of BaseModel.load messages key
+        :returns: formatted string
+
+        """
+        res = []
+        for msg in messages:
+            rows = msg.get('rows', {})
+            res.append(_("%s. -- Field: %s -- rows %s to %s") % (
+                msg.get('message', 'N/A'),
+                msg.get('field', 'N/A'),
+                rows.get('from', 'N/A'),
+                rows.get('to', 'N/A'))
+            )
+        return "\n \n".join(res)
+
+    def _parse_csv(self):
+        """Parse stored CSV file in order to be usable by BaseModel.load method.
+
+        Manage base 64 decoding.
+
+        :param imp_id: current importer id
+        :returns: (head [list of first row], data [list of list])
+
+        """
+        # We use tempfile in order to avoid memory error with large files
+        with tempfile.TemporaryFile() as src:
+            imp = self.read(['file'])
+            content = imp[0]['file']
+            delimiter = '\t'
+            src.write(content)
+            with tempfile.TemporaryFile() as decoded:
+                src.seek(0)
+                base64.decode(src, decoded)
+                decoded.seek(0)
+                return self._prepare_csv_data(decoded, delimiter)
+
+    def _prepare_csv_data(self, csv_file, delimiter=","):
+        """Parse a decoded CSV file and return head list and data list
+
+        :param csv_file: decoded CSV file
+        :param delimiter: CSV file delimiter char
+        :returns: (head [list of first row], data [list of list])
+
+        """
+        try:
+            data = csv.DictReader(csv_file, fieldnames=self.HEAD_CRESUS,
+                                  delimiter=delimiter)
+        except csv.Error as error:
+            raise orm.except_orm(
```

Comment:
`orm.except_orm` is deprecated use instead:

    from openerp import models, [...], exceptions
    [...]
    raise exceptions.Warning(_('message'))